### ISSUE TITLE
kube-state-metrics namespace is hardcoded to "monitoring"

### ISSUE THREAD

**Issue by bgagnon** [coreos/kube-prometheus#460]
This is a regression introduced in #404.

From `kube-state-metrics/kube-state-metrics.libsonnet`:

```jsonnet
  kubeStateMetrics+:: (import 'kube-state-metrics/kube-state-metrics.libsonnet') +
                      {
                        local ksm = self,
                        name:: 'kube-state-metrics',
                        namespace:: 'monitoring',
```

The namespace here should be `$_config.namespace` (which defaults to `'monitoring'`, but can be changed).

**Comment by brancz** [comment]
You're absolutely right. Would you like to create a PR to fix this?

**Comment by bgagnon** [comment]
Fix available in #462
